dataRetriever/requestHandlers/requestHandler.go

```diff
@@ -124,6 +122,7 @@ func (rrh *resolverRequestHandler) requestByHashes(destShardID uint32, hashes []
 		}
 	}()
 
+	rrh.whiteList.Add(hashes)
```

Comment:
Shoudn't this be added to the goroutine? If `txResolver.RequestDataFromHashArray(batch, rrh.epoch)` does not return an error, you could add that batch to the whitelist, otherwise skip the current batch.

Reply:
I still want to whitelist those hashes in any case. whether the request is failed, a broadcast message might be received and the data has to be accepted.